### PR TITLE
Revert repeat marker musical symbols scale changes

### PR DESCRIPTION
**Pull request by miiizen**
Resolves: #24330
Resolves: #24899

This reverts https://github.com/musescore/MuseScore/pull/21758 and adds some compatibility code for reading 4.4.0-2 files & palettes.

### CODE REVIEW

**Review by oktophonie — APPROVED** [review]

File: src/engraving/rw/read410/tread.cpp [hunk 1]
```diff
@@ -1875,6 +1875,11 @@ void TRead::read(Marker* m, XmlReader& e, ReadContext& ctx)
             e.unknown();
         }
     }
+    const bool needsAdjustMarkerSize = m->score()->mscVersion() == 440 && m->score()->mscoreVersion() != u"4.4.3" && !ctx.pasteMode();
+    const bool isDefualtSize = m->score()->style().styleD(Sid::repeatLeftFontSize) == 11.0;
```
Comment (mike-spa) [inline src/engraving/rw/read410/tread.cpp:1879]: *default (small typo)

File: src/palette/internal/palettecompat.cpp [hunk 2]
```diff
@@ -98,7 +94,22 @@ void PaletteCompat::migrateOldPaletteItemIfNeeded(ElementPtr& element, Score* pa
         newPedal->setEndText(newPedal->propertyDefault(Pid::END_TEXT).value<String>());
 
         element.reset(newPedal);
-        return;
+    }
+
+    element->styleChanged();
+
+    // from 4.4.3 markers have a font size which is not the default, so this must be reset after calling style changed
+    if (item->isMarker() && toMarker(item)->size() == 11) {
```
Comment (mike-spa) [inline src/palette/internal/palettecompat.cpp:102]: I'm not sure I understand what is going on here. Using 11 and 18 like this looks a bit magic numbers to me (same for the TRead code above) as I don't know the context. They should be in a variable (or at least have an explanatory comment).
Speaking of comment: what do you mean that markers have a font size which is not the default? Shouldn't an element inserted into the score from the default palettes have default settings by definition?

**Comment by cbjeukendrup** [comment]
It looks like the Palettes would not automatically benefit from this, or do they?

**Comment by miiizen** [comment]
image
They do - they appear at the correct smaller size (same as 4.3).

**Comment by cbjeukendrup** [comment]
Ah, probably the reason that it works is that the palette items just take the value from the Style, which is the default style in the case of the palettes, which is of course updated by this PR. Then the next question is, doesn't that apply to items from files too and is the whole compat code maybe unnecessary, but I think the answer is no, because files contain the old style value, and the only possibility to update it is after the entire file has been read.

Or... can we do it in `MStyle::read`? The only problem is that we'd need access to the `Score` for the version number, but that doesn't seem to be far away. 
You could try if that works (and makes the per-element compat code unnecessary, which would be a nice simplification). But if it doesn't work or you don't think it's a good idea or you don't want to spend more time on this issue, it's also fine to merge this PR as is, as far as I'm concerned.

**Comment by miiizen** [comment]
I tried that on my first attempt, but the style file is necessarily read before the rest of the score where we find the version number we want. I'm not thrilled about the compat code, but I think it's the easiest/most clear (but certainly not most efficient) way of achieving this.

**Pull request merged**